go-controller/pkg/ovn/gateway.go

```diff
@@ -249,14 +249,23 @@ func (gw *GatewayManager) GatewayInit(
 	hostSubnets []*net.IPNet,
 	l3GatewayConfig *util.L3GatewayConfig,
 	sctpSupport bool,
-	gwLRPIfAddrs, drLRPIfAddrs []*net.IPNet,
+	gwLRPJoinIPs, drLRPIfAddrs []*net.IPNet,
 	externalIPs []net.IP,
 	enableGatewayMTU bool,
 ) error {
 
 	gwLRPIPs := make([]net.IP, 0)
-	for _, gwLRPIfAddr := range gwLRPIfAddrs {
-		gwLRPIPs = append(gwLRPIPs, gwLRPIfAddr.IP)
+	for _, gwLRPJoinIP := range gwLRPJoinIPs {
+		gwLRPIPs = append(gwLRPIPs, gwLRPJoinIP.IP)
```

Comment:
later in the function like like 600, gwLRPIPs is used to add a route:
```
	// FIXME(trozet): if LRP IP is changed, we do not remove stale instances of these routes
	for _, gwLRPIP := range gwLRPIPs {
		lrsr := nbdb.LogicalRouterStaticRoute{
			IPPrefix: gwLRPIP.String(),
			Nexthop:  gwLRPIP.String(),
		}
		
```

It would add a route to .1 which would be wrong, but I guess we are impliclty saved because:

```
		if gw.clusterRouterName != "" {
			err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(gw.nbClient,
				gw.clusterRouterName, &lrsr, p, &lrsr.Nexthop)
			if err != nil {
				return fmt.Errorf("error creating static route %+v in %s: %v", lrsr, gw.clusterRouterName, err)
			}
		}
```
gw.clusterRouterName = "" for layer 2 topologies. It would be nice for you to add a comment there to say this

Reply:
yea but even before my change this part stays the same..
before my change the joinIPs and hostSubnets were being passed together
I changed this to explicitly pass joinIPs and then add the hostsubnet within this func 
so the .1 route was always getting added :( -> that's why I tried to get rid of the .1 all together in this logic but then realised we needed it for the ICMP SNAT and left it as is. I will remove all that in the FUP PR

Reply:
yeah we can handle as follow ups